vectornav/package.xml

```diff
@@ -18,8 +18,10 @@
   <!-- TODO: .msg Hex Constant Support Required -->
   <depend version_gte="1.2.0">rosidl_adapter</depend> 
 
-  <test_depend>ament_lint_auto</test_depend>
-  <test_depend>ament_lint_common</test_depend>
+  <test_depend>ament_cmake_cppcheck</test_depend>
+  <test_depend>ament_cmake_clang_format</test_depend>
+  <test_depend>ament_cmake_lint_cmake</test_depend>
+  <test_depend>ament_cmake_xmllint</test_depend>
```

Comment:
```suggestion
  <exec_depend>ament_cmake_cppcheck</exec_depend>
  <exec_depend>ament_cmake_cpplint</exec_depend>
  <exec_depend>ament_cmake_lint_cmake</exec_depend>
  <exec_depend>ament_cmake_xmllint</exec_depend>
```

Can we use `ament_cmake_cpplint` and the original code would pass? `cpp_lint` is the one shipped with [`ament_lint_common` ](https://github.com/ament/ament_lint/blob/rolling/ament_lint_common/package.xml) and I am happy to stick to the default.

Anyhow, I think linting is quite broken altogether. In that case, I think we should not worry about lining and disabling the failing linting tests. This will make it easier to cherry-pick or include features from upstream later. What do you think?

Reply:
The ROS code passes just fine, the vendor library for the IMU is another story, it needs a fair bit of work to get up to scratch:

Category 'build/header_guard' errors found: 70
Category 'build/include_order' errors found: 6
Category 'build/include_subdir' errors found: 66
Category 'build/include_what_you_use' errors found: 6
Category 'build/namespaces' errors found: 30
Category 'legal/copyright' errors found: 59
Category 'readability/braces' errors found: 4
Category 'readability/casting' errors found: 140
Category 'readability/todo' errors found: 24
Category 'runtime/arrays' errors found: 2
Category 'runtime/indentation_namespace' errors found: 127
Category 'runtime/int' errors found: 3
Category 'runtime/printf' errors found: 151
Category 'whitespace/comments' errors found: 330
Category 'whitespace/line_length' errors found: 378
Category 'whitespace/newline' errors found: 1
Category 'whitespace/tab' errors found: 12
Total errors found: 1409

Given we might not be using this long term I wouldn't suggest spending too much time on this, a quick lot at the code and I got a few more grey hairs so I think it might take a while to get up to standard. 

If you're ok lets keep cpp_lint disabled? I've already run it manually on the ROS side and all it passing.